benchmark/bench_dom_api.cpp

```diff
@@ -365,11 +356,12 @@ BENCHMARK(twitter_image_sizes);
 static void error_code_twitter_count(State& state) noexcept {
   // Prints the number of results in twitter.json
   dom::parser parser;
-  auto [doc, error1] = parser.load(TWITTER_JSON);
-  if (error1) { return; }
+  simdjson::error_code error;
+  dom::element doc;
+  if (!parser.load(TWITTER_JSON).get(doc, error)) { return; }
   for (UNUSED auto _ : state) {
-    auto [value, error] = doc["search_metadata"]["count"].get<uint64_t>();
-    if (error) { return; }
+    uint64_t value;
+    if (!doc["search_metadata"]["count"].get(value, error)) { return; }
     if (value != 100) { return; }
```

Comment:
Let us compare here... We had....

```C++
auto [value, error] = doc["search_metadata"]["count"].get<uint64_t>();
if (error) { return; }
```

and we are moving to...

```C++
uint64_t value;
if (!doc["search_metadata"]["count"].get(value, error)) { return; }
```

Reply:
Re-comparing, we had:

```c++
auto [value, error] = doc["search_metadata"]["count"].get<uint64_t>();
if (error) { return; }
```

And are now moving to:

```c++
uint64_t value;
if ((error = doc["search_metadata"]["count"].get(value)) { return; }
```

Or, if you don't like stuff in your if statement:

```c++
uint64_t value;
auto error = doc["search_metadata"]["count"].get(value);
if (error) { return; }
```

It's kind of nice to be able to read the types of important variables like uint64_t on the left hand side when you are scanning. I am finding myself turning against the auto [value, error] syntax in favor of this :) What are your thoughts?